### PR TITLE
Add prism and prismRotation fields to MovingHeadParams.

### PR DESCRIPTION
**Pull request by kcking**
quick demo: 

https://github.com/baaahs/sparklemotion/assets/1348691/57998ee8-9e2f-4a87-8d9e-0d36943af685

### CODE REVIEW

File: src/commonMain/kotlin/baaahs/model/MovingHead.kt [hunk 1]
```diff
@@ -125,6 +145,8 @@ class MovingHead(
         /** Rotation of color wheel in `(0..1]`. */
         var colorWheelPosition: Float
         var shutter: Int
+        var prism: Float
```
Comment (xian) [inline src/commonMain/kotlin/baaahs/model/MovingHead.kt:148]: Boolean?

**Review by xian — APPROVED** [review]
Nice. Just promise to use your powers for good!

File: src/commonMain/kotlin/baaahs/model/MovingHead.kt [hunk 2]
```diff
@@ -111,6 +114,23 @@ class MovingHead(
         override var shutter: Int
             get() = dmxBuffer[adapter.shutterChannel].toInt()
             set(value) { dmxBuffer[adapter.shutterChannel] = value.toByte() }
+
+        override var prism: Float
```
[the remark this answers is not in the document]
Comment (kcking) [reply src/commonMain/kotlin/baaahs/model/MovingHead.kt]: Yes it should!

File: src/commonMain/kotlin/baaahs/gl/glsl/GlslType.kt [hunk 3]
```diff
@@ -89,6 +91,20 @@ sealed class GlslType constructor(
                     append(" }")
                 }.toString().let { GlslExpr(it) }
         }
+
+        override fun outputRepresentationGlsl(varName: String): String{
+            if (outputRepresentationOverride != null) {
+                return outputRepresentationOverride!!(varName);
+            }
+            val buf = StringBuilder()
```
[the remark this answers is not in the document]
Comment (kcking) [reply src/commonMain/kotlin/baaahs/gl/glsl/GlslType.kt]: Thanks!

**Comment by synox** [comment]
Oooooh yeah! 💯

**Pull request merged**